src/libraries/System.Runtime.Extensions/tests/System/IO/Path.RemoveRedundantSegments.cs

```diff
@@ -0,0 +1,503 @@
+
+        // Normal paths
+        public static TheoryData<string, string> WindowsNormalData => new TheoryData<string, string>
+        {
+            // A '\' inside a string that is prefixed with @ is actually passed as an escaped backward slash "\\"
+            { @"C:",               @"C:" },
+            { @"C:\",              @"C:\" },
+            { @"C:\Users",         @"C:\Users" },
+            { @"C:\Users\",        @"C:\Users\" },
+            { @"C:\Users\myuser",  @"C:\Users\myuser" },
+            { @"C:\Users\myuser\", @"C:\Users\myuser\" },
+        };
+        public static TheoryData<string, string> UnixNormalData => new TheoryData<string, string>
+        {
+            // AltDirectorySeparatorChar gets normalized to DirectorySeparatorChar
+            { @"/",                @"\" },
+            { @"/home",            @"\home" },
+            { @"/home/",           @"\home\" },
+            { @"/home/myuser",     @"\home\myuser" },
+            { @"/home/myuser/",    @"\home\myuser\" },
+        };
+
+        // Paths with '..' to indicate parent backtracking
+        public static TheoryData<string, string> WindowsValidParentBacktrackingData => new TheoryData<string, string>
+        {
+            { @"C:\..",                        @"C:\" },
+            { @"C:\..\",                       @"C:\" },
+            { @"C:\..\Users",                  @"C:\Users" },
+            { @"C:\..\Users\",                 @"C:\Users\" },
+            { @"C:\Users\..",                  @"C:\" },
+            { @"C:\Users\..\",                 @"C:\" },
+            { @"C:\Users\..\..",               @"C:\" },
+            { @"C:\Users\..\..\",              @"C:\" },
+            { @"C:\Users\..\myuser",           @"C:\myuser" },
+            { @"C:\Users\..\myuser\",          @"C:\myuser\" },
+            { @"C:\Users\..\..\myuser",        @"C:\myuser" },
+            { @"C:\Users\..\..\myuser\",       @"C:\myuser\" },
+            { @"C:\Users\myuser\..",           @"C:\Users" },
+            { @"C:\Users\myuser\..\",          @"C:\Users\" },
+            { @"C:\Users\myuser\..\..",        @"C:\" },
+            { @"C:\Users\myuser\..\..\",       @"C:\" },
+            { @"C:\Users\..\myuser\..",        @"C:\" },
+            { @"C:\Users\..\myuser\..\",       @"C:\" },
+            { @"C:\Users\..\myuser\..\..",     @"C:\" },
+            { @"C:\Users\..\myuser\..\..\",    @"C:\" },
+            { @"C:\Users\..\myuser\..\..\..",  @"C:\" },
+            { @"C:\Users\..\myuser\..\..\..\", @"C:\" },
+            { @"C:\Users\..\..\myuser\..\..",  @"C:\" },
+            { @"C:\Users\..\..\myuser\..\..\", @"C:\" },
+        };
+        public static TheoryData<string, string> UnixValidParentBacktrackingData => new TheoryData<string, string>
+        {
+            { @"/..",                       @"/" }, // This case is not normalizing the root AltDirectorySeparatorChar because no separator is found after the root (which was ignored)
```

Comment:
@JeremyKuhne - I'm investigating these tests where I put a comment next to them, because their behavior was slightly different than expected.

I also noticed the resulting path is different in Windows than in Unix, because of the separator char used.

Reply:
I talked with Jeremy about the expected behavior on each edge case. I am adding tests to verify them.